marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/matcher-impl/options-impl.xqy

```diff
@@ -578,6 +584,40 @@ declare function opt-impl:compile-match-options(
     )
 };
 
+declare function opt-impl:convert-match-rule-for-custom-module($match-rule, $match-options, $custom-algorithm)
```

Comment:
Is it possible to declare types for these inputs? I always like to see those in XQuery functions, unless the args can be anything (although even item() is useful to signify whether one or multiple are expected).

Reply:
The XQuery functions in SM *should* have declared types, but where do we start? Do we have test coverage to handle all the possible inputs to SM (3 flavors of config, custom modules in XQ and JS) to catch type errors now instead of our customers seeing errors later?

Reply:
Yeah, that's fair.